### ISSUE TITLE
Colyseus 0.14 "refId" not found: 1

### ISSUE THREAD

**Issue by doorbash** [colyseus/colyseus#366]
Hi,

This is a bug regarding `colyseus 0.14` 

When client connects to an already created and running room, server sends a `ROOM_STATE_PATCH(15)` instead of a `ROOM_STATE(14)` message after handshake. and then sever crashes.

## Steps to Reproduce (for bugs)
Run this server: https://github.com/doorbash/temp-server
port number: `2222`. room name: `ffa`
Now connect with any client using `joinOrCreate` 

```typescript
import { Client } from ".";

const client = new Client("ws://localhost:2222");

(async () => {
    try {
        const room = await client.joinOrCreate<any>("ffa");
        
        console.log("joined successfully", room.sessionId);

    } catch (e) {
        console.error("couldn't join room:", e);
    }
})()
```

Then disconnect and connect again you will get : `refId not found`

## Your Environment
* Colyseus Version: `0.14.0-alpha`
* Node.js Version: `v11.10.1`
* Operating System and version: `Windows 10`

For instance this is the data server sends when the the first time client creates the room(after handshake):

```
14,128,1,129,2,130,1,131,0,132,89,221,91,184,116,1,0,0,133,0,0,0,0,0,0,0,0,255,1,128,0,169,49,89,48,90,49,65,78,113,85,3,255,2,128,0,163,99,95,48,4,128,1,163,98,95,49,5,255,3,128,1,0,0,0,129,0,0,48,194,130,0,0,0,0,131,10,43,18,64,132,10,43,18,64,134,6,135,7,136,8,137,0,140,0,0,141,0,0,142,7,0,143,169,35,54,101,99,54,102,102,102,102,144,169,35,50,49,57,54,102,51,102,102,145,0,0,133,169,49,89,48,90,49,65,78,113,85,139,165,103,117,101,115,116,255,4,128,0,0,0,0,129,0,0,0,0,130,1,255,5,128,0,0,143,196,129,0,0,0,0,130,2,255,6,255,7,128,0,169,107,45,50,48,48,48,48,45,49,9,128,1,169,107,45,49,48,48,48,48,45,49,10,128,2,168,107,45,50,48,48,48,48,48,11,128,3,168,107,45,49,48,48,48,48,48,12,128,4,163,107,48,48,13,128,5,168,107,45,50,48,48,48,48,49,14,128,6,168,107,45,49,48,48,48,48,49,15,255,8,255,9,128,254,255,129,255,255,130,1,0,0,0,255,10,128,255,255,129,255,255,130,1,0,0,0,255,11,128,254,255,129,0,0,130,1,0,0,0,255,12,128,255,255,129,0,0,130,1,0,0,0,255,13,128,0,0,129,0,0,130,1,0,0,0,255,14,128,254,255,129,1,0,130,1,0,0,0,255,15,128,255,255,129,1,0,130,1,0,0,0
```

and this the data server sends when client connects again to room:
```
15,255,1,128,11,169,106,65,108,116,56,86,45,118,119,205,10,15,255,57,131,58,44,215,193,129,71,219,0,68,130,35,92,135,196,255,204,237,131,159,83,34,193,129,55,126,171,67,130,152,67,101,68,255,18,131,55,94,72,63,129,9,144,190,68,130,120,237,169,195,255,204,143,131,254,91,180,192,129,175,27,82,196,130,201,97,51,68,255,205,142,1,131,6,154,185,193,129,234,251,157,196,130,139,8,233,66,255,205,41,3,131,126,134,108,63,129,148,0,199,68,130,72,57,135,196,255,205,246,3,131,41,224,125,193,129,145,110,141,196,130,206,202,124,196,255,205,61,2,131,219,166,203,65,129,169,198,233,66,130,167,63,241,196,132,232,160,1,63,255,205,41,5,131,214,136,179,65,129,227,91,172,68,130,241,72,132,68,255,205,77,6,131,63,223,99,193,129,201,236,9,69,130,11,226,251,66,132,70,115,16,192,255,205,10,15,128,12,0,0,0,129,0,128,159,196,130,0,128,161,68,131,214,201,13,62,132,214,201,13,62,134,205,11,15,135,205,12,15,136,205,13,15,137,0,140,0,0,141,0,0,142,7,0,143,169,35,102,101,50,99,100,52,102,102,144,169,35,97,53,49,102,56,97,102,102,145,0,0,133,169,106,65,108,116,56,86,45,118,119,139,165,103,117,101,115,116,255,4,128,108,29,191,196,129,12,237,32,68,255,5,128,9,53,146,196,129,115,86,207,68,255,17,128,14,254,162,196,129,155,52,241,196,255,29,128,34,100,172,196,129,167,113,213,68,255,30,128,62,236,71,68,129,16,246,164,196,255,42,128,10,12,177,68,129,160,247,142,196,255,43,128,174,61,15,197,129,174,99,242,196,255,70,128,70,114,22,69,129,227,37,214,67,255,71,128,39,240,20,69,129,91,71,96,195,255,96,128,254,217,42,197,129,217,189,125,68,255,204,156,128,247,161,130,196,129,214,6,36,68,255,204,157,128,225,58,23,197,129,233,119,12,68,255,204,199,128,225,207,161,68,129,60,189,147,68,255,204,200,128,214,8,109,196,129,154,189,243,68,255,204,252,128,190,104,43,196,129,144,186,52,68,255,204,253,128,179,138,97,68,129,70,28,72,196,255,205,163,1,128,76,3,15,68,129,75,195,154,195,255,205,164,1,128,189,225,123,68,129,183,235,131,67,255,205,245,1,128,128,245,123,67,129,46,208,163,68,255,205,246,1,128,28,40,183,196,129,67,205,116,67,255,205,81,2,128,14,236,170,196,129,252,28,111,196,255,205,82,2,128,115,32,151,67,129,221,62,57,196,255,205,198,2,128,223,106,19,196,129,47,132,101,68,255,205,199,2,128,179,251,117,68,129,13,81,191,66,255,205,89,3,128,42,108,56,68,129,231,70,17,196,255,205,90,3,128,22,215,176,195,129,211,232,220,195,255,205,183,3,128,184,177,247,68,129,237,225,241,68,255,205,45,4,128,254,131,8,196,129,78,199,21,197,255,205,46,4,128,124,44,18,196,129,27,143,24,68,255,205,246,4,128,243,213,3,197,129,76,139,68,68,255,205,83,5,128,113,178,121,67,129,139,230,42,195,255,205,84,5,128,64,11,3,197,129,241,169,120,68,255,205,205,5,128,120,203,185,68,129,118,147,48,68,255,205,206,5,128,233,47,1,69,129,136,10,233,196,255,205,116,6,128,227,185,202,195,129,16,148,128,68,255,205,117,6,128,154,33,160,196,129,45,136,58,195,255,205,77,7,128,79,117,115,196,129,182,154,157,193,255,205,78,7,128,64,99,148,68,129,111,92,116,196,255,205,40,8,128,203,31,89,196,129,86,203,251,68,255,205,216,8,128,77,214,246,67,129,229,242,154,68,255,205,217,8,128,25,164,147,196,129,74,99,201,67,255,205,162,9,128,91,189,173,195,129,94,210,204,196,255,205,163,9,128,254,178,18,197,129,185,167,13,196,255,205,82,10,128,180,167,160,67,129,134,183,143,196,255,205,83,10,128,16,111,93,196,129,45,75,99,196,255,205,30,11,128,234,233,184,196,129,91,208,238,195,255,205,31,11,128,110,206,67,68,129,193,138,183,196,255,205,132,11,128,92,114,118,67,129,29,49,134,196,255,205,133,11,128,103,247,39,68,129,187,27,230,68,255,205,242,11,128,104,97,241,68,129,127,171,175,194,255,205,243,11,128,246,225,120,196,129,129,45,182,196,255,205,237,12,128,101,152,54,67,129,44,135,152,67,255,205,238,12,128,199,188,250,68,129,71,3,121,68,255,205,57,13,128,27,95,255,196,129,120,113,23,196,255,205,58,13,128,122,0,201,194,129,229,39,48,68,255,205,182,13,128,139,142,22,68,129,76,240,176,68,255,205,183,13,128,138,205,215,196,129,169,49,155,194,255,205,91,14,128,75,69,167,196,129,29,221,139,68,255,205,92,14,128,24,184,155,194,129,169,14,16,196,255,58,128,204,169,205,14,15,255,19,128,204,183,205,15,15,255,204,146,128,102,205,16,15,255,204,144,128,204,155,205,17,15,255,205,62,2,128,127,205,18,15,255,205,80,6,128,61,205,19,15,255,205,78,6,128,82,205,20,15,255,205,12,15,128,0,170,107,45,51,48,48,48,48,48,51,51,205,21,15,128,1,170,107,45,50,57,48,48,48,48,51,51,205,22,15,128,2,170,107,45,51,48,48,48,48,48,51,52,205,23,15,128,3,170,107,45,50,57,48,48,48,48,51,52,205,24,15,128,4,170,107,45,50,56,48,48,48,48,51,52,205,25,15,128,5,170,107,45,51,48,48,48,48,48,51,53,205,26,15,128,6,170,107,45,50,57,48,48,48,48,51,53,205,27,15,255,205,14,15,128,71,219,0,68,129,35,92,135,196,255,205,15,15,128,9,144,190,68,129,120,237,169,195,255,205,16,15,128,236,255,129,19,0,130,4,0,0,0,255,205,17,15,128,175,27,82,196,129,201,97,51,68,255,205,18,15,128,169,198,233,66,129,0,128,237,196,255,205,19,15,128,50,0,129,3,0,130,11,0,0,0,255,205,20,15,128,201,236,9,69,129,11,226,251,66,255,205,21,15,128,226,255,129,33,0,130,12,0,0,0,255,205,22,15,128,227,255,129,33,0,130,12,0,0,0,255,205,23,15,128,226,255,129,34,0,130,12,0,0,0,255,205,24,15,128,227,255,129,34,0,130,12,0,0,0,255,205,25,15,128,228,255,129,34,0,130,12,0,0,0,255,205,26,15,128,226,255,129,35,0,130,12,0,0,0,255,205,27,15,128,227,255,129,35,0,130,12,0,0,0
```

**Comment by endel** [comment]
Hi @doorbash, thanks a lot for reporting, and for providing how to reproduce it!

I've just released `@colyseus/schema@1.0.0-alpha.55` with a fix for this! Let me know if you find anything else!

**Comment by doorbash** [comment]
Hey @endel Thank you for fixing this so quickly!. It was a pain in the ass. You're awesome 🤟
